convert-md-to-json.ts

```diff
@@ -0,0 +1,96 @@
+/// <reference types="node" />
+/**
+ * This is the script for converting old .md files into new data-oriented scheme.
+ *
+ * It's intended to be used once, and removed after migration.
+ *
+ * When preparing migration into master, use:
+ *
+ *   git checkout master -- content/components-and-libraries/
+ *   tsx convert-md-to-json.ts
+ *   npm run fetch-data
+ *
```

Comment:
Now, it's ready to be merged.

I converted all md-files into data-oriented format.

I think, we may have need this file one last time when merging into master.

@rmjordas please, take a look!

Reply:
This is great, @last-partizan! My initial thought is: Right now, this proposal would probably break the "Edit this page in GitHub" feature at the bottom of the page. 

If someone external to the project clicks this link and they wanted to add an entry, it will take them to the markdown files. However, they wouldn't be able to insert their link entry because the markdown items have been replaced by a `<ProjectList>` component.

We could remove this edit link or add more instructions for contributors or maybe something else. But first, I'm going to experiment with your proposal and figure out if the caching, conversion, and fetching are can be done while building the website. If this is the case, I'm hoping we might be able to preserve the original markdown files and there's no need to commit the `*.json` and `*.js` files to the repository. 

I know this will probably strech the build times but maybe not that much. We'll see. And also, let me know if you have solutions for this!

Give me a few days/weeks to review. 🙏

Reply:
This is great, @last-partizan! My initial thought is: Right now, this proposal would probably break the "Edit this page in GitHub" feature at the bottom of the page. 

If someone external to the project clicks this link and they wanted to add an entry, it will take them to the markdown files. However, they wouldn't be able to insert their link entry because the markdown items have been replaced by a `<ProjectList>` component.

We could remove this edit link or add more instructions for contributors or maybe something else. But first, I'm going to experiment with your proposal and figure out if the caching, conversion, and fetching are can be done while building the website. If this is the case, I'm hoping we might be able to preserve the original markdown files and there's no need to commit the `*.json` and `*.js` files to the repository. 

I know this will probably strech the build times but maybe not that much. We'll see. And also, let me know if you have solutions for this!

Give me a few days/weeks to review. 🙏

Reply:
I think, it's best to store data in structured format, because parsing it from markdown each time is more error-prone.

And, if we add clear instructions on how to add new data - it would be fine. Maybe we can change "Edit this page" link to point to data file, instead of md.

I've already found one unicode mdash instead of - :)

Reply:
Just came here via a discussion at Vue.js Nation that mentioned Awesome Vue.

Have you considered storing data in the frontmatter? https://vitepress.dev/guide/frontmatter

Reply:
It is certainly possible to store data in the frontmatter, but i don't know what benefits of this approach.

frontmatter is better suited for storing small pieces of data, like page title and metadata. But here we trying to write data back to json files, with added remote-fetched details.

It's easier to do with separate json files.